wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -0,0 +1,1587 @@
+                                        ConfigConstants.CONFIG_KEY_REMOTE);
+            if (name.equals(r)) {
+                config.unset(ConfigConstants.CONFIG_BRANCH_SECTION, branch, ConfigConstants.CONFIG_KEY_REMOTE);
+                config.unset(ConfigConstants.CONFIG_BRANCH_SECTION, branch, ConfigConstants.CONFIG_KEY_MERGE);
+                List<Branch> remoteBranches = branchList(newDto(BranchListRequest.class).withListMode("r"));
+                for (Branch remoteBranch : remoteBranches) {
+                    if (remoteBranch.getDisplayName().startsWith(name)) {
+                        branchDelete(newDto(BranchDeleteRequest.class).withName(remoteBranch.getName()).withForce(true));
+                    }
+                }
+            }
+        }
+
+        try {
+            config.save();
+        } catch (IOException exception) {
+            throw new GitException(exception.getMessage(), exception);
+        }
+    }
+
+    @Override
+    public List<Remote> remoteList(RemoteListRequest request) throws GitException {
+        StoredConfig config = repository.getConfig();
+        Set<String> remoteNames = new HashSet<>(config.getSubsections(ConfigConstants.CONFIG_KEY_REMOTE));
+        String remote = request.getRemote();
+
+        if (remote != null && remoteNames.contains(remote)) {
+            remoteNames.clear();
+            remoteNames.add(remote);
+        }
+
+        List<Remote> result = new ArrayList<>(remoteNames.size());
+        for (String remoteName : remoteNames) {
+            try {
+                List<URIish> uris = new RemoteConfig(config, remoteName).getURIs();
+                result.add(newDto(Remote.class).withName(remoteName).withUrl(uris.isEmpty() ? null : uris.get(0).toString()));
+            } catch (URISyntaxException exception) {
+                throw new GitException(exception.getMessage(), exception);
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public void remoteUpdate(RemoteUpdateRequest request) throws GitException {
+        String remoteName = request.getName();
+        if (isNullOrEmpty(remoteName)) {
+            throw new IllegalArgumentException(ERROR_UPDATE_REMOTE_NAME_MISSING);
+        }
+
+        StoredConfig config = repository.getConfig();
+        Set<String> remoteNames = config.getSubsections(ConfigConstants.CONFIG_KEY_REMOTE);
+        if (!remoteNames.contains(remoteName)) {
+            throw new IllegalArgumentException("Remote " + remoteName + " not found. ");
+        }
+
+        RemoteConfig remoteConfig;
+        try {
+            remoteConfig = new RemoteConfig(config, remoteName);
+        } catch (URISyntaxException e) {
+            throw new GitException(e.getMessage(), e);
+        }
+
+        List<String> branches = request.getBranches();
+        if (!branches.isEmpty()) {
+            if (!request.isAddBranches()) {
+                remoteConfig.setFetchRefSpecs(new ArrayList<>());
```

Comment:
the two following empty lists can be turned into Collections.emptyList()
as remoteConfig only add elements
http://grepcode.com/file/repo1.maven.org/maven2/org.eclipse.jgit/org.eclipse.jgit/3.3.0.201403021825-r/org/eclipse/jgit/transport/RemoteConfig.java#RemoteConfig.setFetchRefSpecs%28java.util.List%29

I am also wondering if we require to call these two setters with empty stuff